react/features/toolbox/components/AudioRouteButton.js

```diff
@@ -0,0 +1,153 @@
+// @flow
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import {
+    findNodeHandle,
+    requireNativeComponent,
+    NativeModules,
+    Platform,
+    View
+} from 'react-native';
+import { connect } from 'react-redux';
+
+import ToolbarButton from './ToolbarButton';
+
+import { openDialog } from '../../base/dialog';
+import { AudioRoutePickerDialog } from '../../mobile/audio-mode';
+
+/**
+ * Define the {@code MPVolumeView} React component. It will only be available
+ * on iOS.
+ */
+const MPVolumeView = requireNativeComponent('MPVolumeView', null);
+
+/**
+ * Style required to hide the {@code MPVolumeView} view, since it's displayed
+ * programmatically.
+ */
+const HIDE_VIEW_STYLE = { display: 'none' };
+
+/**
+ * A toolbar button which triggers an audio route picker when pressed.
+ */
+class AudioRouteButton extends Component<*> {
+    /**
+     * AudioRouteButton component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * TODO.
+         */
+        dispatch: PropTypes.func,
+
+        /**
+         * The name of the Icon of this {@code AudioRouteButton}.
+         */
+        iconName: PropTypes.string,
+
+        /**
+         * The style of the Icon of this {@code AudioRouteButton}.
+         */
+        iconStyle: PropTypes.object,
+
+        /**
+         * {@code AudioRouteButton} styles.
+         */
+        style:
+            PropTypes.oneOfType([
+                PropTypes.array,
+                PropTypes.object
+            ]),
+
+        /**
+         * The color underlying the button.
+         */
+        underlayColor: PropTypes.any
+    };
+
+    _volumeComponent: ?Object;
+
+    /**
+     * Initializes a new {@code AudioRouteButton} instance.
+     *
+     * @param {Object} props - The React {@code Component} props to initialize
+     * the new {@code AudioRouteButton} instance with.
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * The internal reference to the React {@code MPVolumeView} for
+         * showing the volume control view.
+         *
+         * @private
+         * @type {ReactComponent}
+         */
+        this._volumeComponent = null;
+
+        // Bind event handlers so they are only bound once per instance.
+        this._onClick = this._onClick.bind(this);
+        this._setVolumeComponent = this._setVolumeComponent.bind(this);
+    }
+
+    _onClick: () => void;
+
+    /**
+     * Handles clicking/pressing this {@code AudioRouteButton} by showing an
+     * audio route picker.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onClick() {
+        if (Platform.OS === 'ios') {
```

Comment:
* We don't seem to need the `Platform.OS` checks here. The question about the platform has already been answered by, for example, the availability of `NativeModules.MPVolumeViewManager`.

* Which leads to: Why `AudioRoutePickerDialog` exists for both iOS and Android? Why isn't `AudioRoutePickerDialog` available only when `AudioMode.getAudioDevices` is a defined method?

Reply:
I'll fix that.

Reply:
Force-pushed with a fix. PTAL at the approach and let me know if you like it that way. I also went ahead and added a supported() static method so the toolbar can check if it can render the button.